doc/source/train/user-guides/experiment-tracking.rst

```diff
@@ -1,195 +1,340 @@
+    Ray Train lets you apply logic to only the rank 0 worker with the following method:
+    :meth:`context.get_world_rank() <ray.train.context.TrainContext.get_world_rank>`.
+
+Step 1: Set up necessary components to be able to connect to the tracking backend of your choice
+------------------------------------------------------------------------------------------------
+
+First, you should choose which tracking backend to use: W&B, MLflow, TensorBoard etc.
+
+Some of them offer to operate under either online or offline mode. They have different considerations when
+setting up.
+For online mode, you log towards a tracking service that is running. Usually you need credentials to access the service.
+Under this mode, you need to ensure that all nodes and worker processes have access to credentials.
+For offline mode, you log towards local file directory. Usually no credentials are needed. You need to instead
+ensure that there is a shared file system where all nodes can write to.
+
+.. tabs::
+
+    .. tab:: Wandb
+
+        .. tabs::
+
+            .. tab:: online mode
+
+                Make sure that :code:`wandb.login(key="your_api_key")` 
+                or :code:`os.environ["WANDB_API_KEY"] = "your_api_key"` is called inside your ``train_func``.
+
+            .. tab:: offline mode
+
+                Make sure that :code:`os.environ["WANDB_MODE"] = "offline"` is set in ``train_func``.
+
+                Set Wandb directory to point to a shared storage path: :code:`wandb.init(dir="some_shared_storage_path/wandb")` 
+
+    .. tab:: MLflow
+
+        .. tabs::
+
+            .. tab:: online mode (hosted by Databricks)
+                
+                Start the run with :code:`mlflow.start_run(tracking_uri="databricks")`
+
+                Make sure that all nodes have access to ``databrickscfg`` file.
+
+            .. tab:: offline mode
 
+                Start the run by setting tracking uri to a shared storage path: 
+                :code:`mlflow.start_run(tracking_uri="file:some_shared_storage_path/mlruns")`
+
+    .. tab:: TensorBoard (offline)
+        
+        Set up ``SummaryWriter`` to write to a shared storage path: :code:`writer = SummaryWriter("some_shared_storage_path/runs")`
+
+Step 2: Initialize the run 
+--------------------------
+
+Ray Train provides a training context that provides access to training identifiers. For example, 
+
+* Training ID (:meth:`context.get_trial_id() <ray.train.context.TrainContext.get_trial_id>`) 
+* Training Name (:meth:`context.get_trial_name() <ray.train.context.TrainContext.get_trial_name>`)
+
+.. tip::
+    
+    When performing **fault-tolerant training** with auto-restoration, be sure 
+    to specify a unique ID for the Loggers, so that the new workers report to
+    the same run after restoration.
+
+    For example:
+    
+    - `WandbLogger(id=UNIQUE_ID)`
+    - `CometLogger(experiment_key=UNIQUE_ID)`
+    - `MLFlowLogger(run_id=UNIQUE_ID)`
```

Comment:
Similarly, are these referencing the Training ID or something else? As a user how do I tie this with the above?

Can I use the trial ID here?

Reply:
The `UNIQUE_ID` doesn't have to be Training ID, it can be anything that's unique tbh...
I sense this is confusing to people. Maybe we should remove the suggestions on what can go in as `UNIQUE_ID`?
It really can simply just `uuid.uuid4()`.